website/content/docs/troubleshoot/troubleshoot-services.mdx

```diff
@@ -0,0 +1,136 @@
+
+For more information about using Envoy proxies with Consul, refer to [Envoy proxy configuration for service mesh](/consul/docs/connect/proxies/envoy).
+
+## Requirements
+
+- Consul v1.15 or later.
+- For Kubernetes, the `consul-k8s` CLI must be installed.
+
+### Technical constraints
+
+When troubleshooting service-to-service communication issues, be aware of the following constraints:
+
+- The troubleshooting tool does not check service intentions. For more information about intentions, including precedence and match order, refer to [service mesh intentions](/consul/docs/connect/intentions).
+- The troubleshooting tool validates one direct connection between a downstream service and an upstream service. You must run the `consul troubleshoot` command with the Envoy ID for an individual upstream service. It does support validating multiple connections simultaneously.
+- Because it validates direct communication between two Envoy proxies, the troubleshooting tool does not support checks for service-to-service connections when communication between the services passes through a mesh gateway or a terminating gateway.
+
+## Usage
+
+Using the service-to-service troubleshooting tool is a two-step process:
+
+1. Find the identifier for the upstream service.
+1. Use the upstream’s identifier to validate communication.
+
+In deployments without transparent proxies, the identifier is the _Envoy ID for the upstream service’s sidecar proxy_. If you use transparent proxies, the identifier is the _upstream service’s IP address_. For more information about using transparent proxies, refer to [Enable transparent proxy mode](/consul/docs/connect/transparent-proxy).
+
+### Troubleshoot on VMs
+
+To troubleshoot service-to-service communication issues in deployments that use VMs or bare-metal servers:
+
+1. Run the `consul troubleshoot upstreams` command to retrieve the upstream information for the service that is experiencing communication failures. Depending on your network’s configuration, the upstream information is either an Envoy ID or an IP address.
+
+```shell-session
+  $ consul troubleshoot upstreams
+
+  ==> Upstreams (explicit upstreams only) (0)
+  ==> Upstreams IPs (transparent proxy only) (1)
+  [10.4.6.160 240.0.0.3] true map[backend.default.dc1.internal.e08fa6d6-e91e-dfe0-f6e1-ba097a828e31.consul backend2.default.dc1.internal.e08fa6d6-e91e-dfe0-f6e1-ba097a828e31.consul]
+  If you cannot find the upstream address or cluster for a transparent proxy upstream:
+  - Check intentions: Tproxy upstreams are configured based on intentions. Make sure you have configured intentions to allow traffic to your upstream.
+  - To check that the right cluster is being dialed, run a DNS lookup for the upstream you are dialing. For example, run `dig backend.svc.consul` to return the IP address for the `backend` service. If the address you get from that is missing from the upstream IPs, it means that your proxy may be misconfigured.
+```
```

Comment:
this isn't properly rendering 

````suggestion
   ```shell-session
   $ consul troubleshoot upstreams

   ==> Upstreams (explicit upstreams only) (0)
   ==> Upstreams IPs (transparent proxy only) (1)
   [10.4.6.160 240.0.0.3] true map[backend.default.dc1.internal.e08fa6d6-e91e-dfe0-f6e1-ba097a828e31.consul backend2.default.dc1.internal.e08fa6d6-e91e-dfe0-f6e1-ba097a828e31.consul]
   If you cannot find the upstream address or cluster for a transparent proxy upstream:
   - Check intentions: Tproxy upstreams are configured based on intentions. Make sure you have configured intentions to allow traffic to your upstream.
   - To check that the right cluster is being dialed, run a DNS lookup for the upstream you are dialing. For example, run `dig backend.svc.consul` to return the IP address for the `backend` service. If the address you get from that is missing from the upstream IPs, it means that your proxy may be misconfigured.
   ```
````

<img width="730" alt="image" src="https://user-images.githubusercontent.com/7556325/220872235-011a8a95-4f72-45f2-b73f-7c552682e895.png">